src/lib.rs

```diff
@@ -107,9 +107,10 @@ pub fn discover_sources(
             has_config_files.get(&meta.name).map(|configs| RepoInfo {
                 repo_name: meta
                     .repo_url
-                    .as_ref()
-                    .map(|url| url.rsplit_once('/').unwrap().1.to_owned())
-                    .expect("errored before now if url was malformed"),
+                    .as_deref()
+                    .and_then(repo_name_from_url)
+                    .expect("already checked")
```

Comment:
I'm using expect here as basically an alternative to a comment; this be unreachable because in order to get here we already called this function once on this input and succeeded.